src/commands/command-fns/thanks.js

```diff
@@ -55,11 +107,86 @@ Link: <${messageLink}>
   )
   return result
 }
+
+async function thanks(message) {
+  const args = getCommandArgs(message.content)
+  const rankArgs = args.replace(MessageMentions.USERS_PATTERN, '').trim()
+
+  let thanksHistory
+  try {
+    thanksHistory = await getThanksHistory()
+  } catch {
+    return message.channel.send(
+      `There is an issue retrieving the history. Please try again later 👎`,
+    )
+  }
+
+  const rankArgumentList = rankArgs.split(' ')
+  if (
+    rankArgumentList.length === 2 &&
+    rankArgumentList[0] === 'rank' &&
+    rankArgumentList[1] === 'top'
+  ) {
+    const sortedUsers = Object.keys(thanksHistory).sort((a, b) => {
+      return thanksHistory[b].length - thanksHistory[a].length
+    })
+    const topUsers = []
+    await sortedUsers.forEach(async user => {
+      const member = await message.guild.members.fetch(user)
+      if (member) {
+        topUsers.push(member.user)
+      }
+    })
+    return message.channel.send(
+      `This is the list of the top thanked members 💪🏼:
+${topUsers
+  .map(
+    user =>
+      `- ${user.username} has been thanked  ${
+        thanksHistory[user.id].length
+      } times 👏`,
+  )
+  .join('\n')}
+`,
+    )
+  } else if (rankArgumentList.length === 1 && rankArgumentList[0] === 'rank') {
+    const mentionedMembers = Array.from(message.mentions.members.values())
+    let searchedMembers = [message.author]
+    if (mentionedMembers.length > 0) {
+      searchedMembers = mentionedMembers.map(member => member.user)
+    }
+
+    return message.channel.send(
+      `This is the rank of the requested member${
+        searchedMembers.length === 1 ? '' : 's'
+      }: 
+${searchedMembers
+  .map(member => {
+    return thanksHistory[member.id]
+      ? `- ${member.username} has been thanked ${
+          thanksHistory[member.id].length
+        } times 👏`
+      : `- ${member.username} hasn't been thanked yet 🙁`
+  })
+  .join('\n')}`,
+    )
```

Comment:
Bit of a refactor (and handle pluralization).

```suggestion
const rankings = searchedMembers
  .map(member => {
    const thanks = thanksHistory[member.id]
    return thanks
      ? `- ${member.username} has been thanked ${thanks.length} time${
          thanks.length === 1 ? '' : 's'
        } 👏`
      : `- ${member.username} hasn't been thanked yet 🙁`
  })
  .join('\n')

message.channel.send(
  `
This is the rank of the requested member${
    searchedMembers.length === 1 ? '' : 's'
  }: 
${rankings}
  `.trim(),
)
```